tests/planner.lua

```diff
@@ -0,0 +1,87 @@
+local TestEZ = script.Parent.Parent.TestEZ
+local TestPlanner = require(TestEZ.TestPlanner)
+local TestBootstrap = require(TestEZ.TestBootstrap)
+local TestEnum = require(TestEZ.TestEnum)
+
+local testRoot = script.Parent.planning
+
+local function verifyPlan(plan, expected, notSkip)
+	local nodes = plan:findNodes(function(node)
+		return (not notSkip) or (node.modifier ~= TestEnum.NodeModifier.Skip)
+	end)
+
+	local nodeNames = {}
+	for _, node in pairs(nodes) do
+		table.insert(nodeNames, node:getFullName())
+	end
+
+	local pass = true
+	local message = ""
+
+	for _, e in pairs(expected) do
+		local ok = false
+		for _, g in pairs(nodeNames) do
+			if e == g then
+				ok = true
+				break
+			end
+		end
+		if not ok then
+			pass = false
+			message = message .. string.format("expected name '%s' not found, ", e)
+		end
+	end
+
+	for _, g in pairs(nodeNames) do
+		local ok = false
+		for _, e in pairs(expected) do
+			if e == g then
+				ok = true
+				break
+			end
+		end
+		if not ok then
+			pass = false
+			message = message .. string.format("additional name '%s' found, ", g)
+		end
+	end
+
+	return pass, message
+end
+
+return {
+	["it should build the full plan with no arguments"] = function()
+		local modules = TestBootstrap:getModules(testRoot)
+		local plan = TestPlanner.createPlan(modules)
+		assert(verifyPlan(plan, {
+			"planning",
+			"planning a",
+			"planning a test1",
+			"planning a test2",
+			"planning b",
+			"planning b test1",
+			"planning b test2",
+			"planning b test2 test3",
+		}))
+	end,
+	["it should mark skipped tests as skipped"] = function()
+		local modules = TestBootstrap:getModules(testRoot)
+		local plan = TestPlanner.createPlan(modules)
+		assert(verifyPlan(plan, {
+			"planning",
+			"planning a",
+			"planning a test2",
+			"planning b",
+			"planning b test1",
+			"planning b test2 test3", -- This isn't marked skip, its parent is
+		}, true))
+	end,
+	["it should skip tests that don't match the filter"] = function()
+		local modules = TestBootstrap:getModules(testRoot)
+		local plan = TestPlanner.createPlan(modules, false, "test2")
+		assert(verifyPlan(plan, {
+			"planning a test2",
+			"planning b test2 test3", -- Gets focus because only its parent is skip
```

Comment:
This test shows that the filter overrides `SKIP`, right? I didn't know that!

Reply:
Well, it shows that the node is set as focused, but I'm not quite sure yet whether or not it will actually get run or if the parent being marked as skipped will stop it there.